bin/inference/pycbc_inference_model_stats

```diff
@@ -92,20 +93,34 @@ model = models.read_from_config(cp)
 # from the variable parameter space
 model.sampling_transforms = None
 
-# create function for calling the model to get the stats
+# create function for calling the model to get the stats, lognl, and rec
 def callmodel(arg):
     iteration, paramvals = arg
     # calculate the logposterior to get all stats populated
     model.update(**{p: paramvals[p] for p in model.variable_params})
     _ = model.logposterior
     stats = model.get_current_stats()
+    if hasattr(model, 'submodels'):
```

Comment:
@WuShichao Wouldn't it make more sense for heirarchical models to make it so that `get_current_stats` includes any submodels rather than hardcoding knowledge about submodels into this script? It's best to keep information compartmentalized and use clearer interfaces to avoid technical debt in the future and prevent flexibility. 

Similarly, why do we need to hardnode anything about lognl here? Maybe it should just be included in the stats when it is a fixed number like this, no?

Reply:
The `lognl` issue is not the hierarchical-model-only problem; whenever users use this script and `lognl` is much larger than `loglr`, the same problem will occur. So instead of modifying all likelihood models, I only change in one place. The reason why I hardcode `lognl` is I want to match the keys in the original PE file.

Reply:
@WuShichao (1) You are explicitly checking through submodels here. That really should be handled by the hierarchical model itself. 

(2) lognl should probably just be a model stat if it is a constant. No modifications are then needed here.